packages/sdk-react/src/components/userRoles/UserRoleDetails/ExistingUserRoleDetails/ExistingUserRoleDetails.tsx

```diff
@@ -0,0 +1,315 @@
+    },
+    {
+      id: PayableActionEnum.CANCEL,
+      headerName: t(i18n)`Cancel`,
+      sortable: false,
+      width: 80,
+      renderCell: renderPermissionCell,
+    },
+    {
+      id: PayableActionEnum.REOPEN,
+      headerName: t(i18n)`Reopen`,
+      sortable: false,
+      width: 80,
+      renderCell: renderPermissionCell,
+    },
+    {
+      id: PayableActionEnum.CREATE_FROM_MAIL,
+      headerName: t(i18n)`Create from mail`,
+      sortable: false,
+      width: 80,
+      renderCell: renderPermissionCell,
+    },
+  ];
+
+  return (
+    <MoniteStyleProvider>
+      <DialogTitle>
+        <Box
+          display="flex"
+          justifyContent="space-between"
+          alignItems="center"
+          gap={2}
+        >
+          <Typography variant="h3" sx={{ wordBreak: 'break-word' }}>
+            {role?.name}
+          </Typography>
+          {dialogContext?.isDialogContent && (
+            <IconButton
+              edge="start"
+              color="inherit"
+              onClick={dialogContext.onClose}
+              aria-label={t(i18n)`Close role details`}
+            >
+              <CloseIcon />
+            </IconButton>
+          )}
+        </Box>
+      </DialogTitle>
+      <Divider />
+      <StyledDialogContainer>
+        <StyledTableTitle variant="subtitle2" mt={4} mb={1}>
+          {t(i18n)`Permissions`}
+          <StyledDocLink
+            underline="none"
+            rel="noopener noreferrer"
+            href="https://docs.monite.com/docs/monite-account-structure#connect-entity-users"
+            target="_blank"
+          >
+            {t(i18n)`Go to Docs`}
+            &nbsp;
+            <OpenInNewIcon />
+          </StyledDocLink>
+        </StyledTableTitle>
+        {role.permissions.objects && (
+          <StyledTableContainer>
+            <Table stickyHeader>
+              <StyledTableHead>
+                <TableRow>
+                  {columns.map((column, index) => (
+                    <StyledTableCell key={column.id}>
+                      {column.headerName}
+                    </StyledTableCell>
+                  ))}
+                </TableRow>
+              </StyledTableHead>
+              <TableBody>
+                {rows.map((row) => {
+                  return (
+                    <TableRow key={row.name}>
+                      {columns.slice(0, 1).map((column) => {
+                        return (
+                          <StyledTableCell key={column.id}>
+                            {row.name &&
+                              getPermissionToLabelMap(i18n)[row.name]}
+                          </StyledTableCell>
+                        );
+                      })}
+                      {columns.slice(1).map((column) => {
+                        const action = column.id as keyof Actions;
+
+                        return (
+                          <StyledTableCell key={column.id}>
+                            {typeof row[action] === 'boolean' ? (
```

Comment:
It is quite challenging to understand this fragment. Can you name it using variables?

Reply:
fixed